docs/bundle_format.rst

```diff
@@ -0,0 +1,106 @@
+Bundle Format
+================
+
+The Blockstore does not parse or validate most files itself. However, some conventions must be followed by clients
+to allow interoperability between environments:
+
+* There are no constraints on how the files in a Bundle can be organized.
+* OLX files must use the **.olx** extension. Files with a different extension may be ignored or not treated as
+  OLX by plugins and clients.
+* A bundle.json file must be added by authoring clients which specifies the shareable components, bundle
+  dependencies and assets. See below for details.
+* Every XBlock element in an OLX file must have an **id** attribute. The value of this attribute will be used
+  to construct the **block_key**. In a single Bundle, all ids must be unique. Blockstore will not validate ids however.
+* Nested XBlocks in separate OLX files can be referenced by adding a **path** attribute to the element which points
+  to the relevant OLX file. Blockstore does not parse or validate any paths itself. However plugins may do so.
+* XBlock Runtimes should provide a service which can be used to read a file given a **path** or to get a
+  public URL for it (if it is specified in assets field). XBlocks will be responsible for replacing paths
+  in any assets and resources they are using with URLs.
+
+Paths
+-------------------------
+
+Paths are URIs of the form **[//bundle_version_alias]path**.
+
+bundle.json
+-------------------------
+The bundle.json file contains metadata about the Bundle. Authoring clients must add and update this themselves.
+It contains the following fields:
+
+meta
+~~~~~~~~~~~~~~~~~~~~~~
+
+.. code-block::
+
+  "meta": {
+    "version": 1
+  },
+
+
+main_component
+~~~~~~~~~~~~~~~~~~~~~~
+
+The optional **main_component** field can be used to specify the file path to an XBlock which is the primary XBlock
+of the Bundle that authors should use.
```

Comment:
Do we need this? Or can just the `components` field do?

Reply:
I think it's fine to have just the `components` field.

Reply:
@ormsbee Great! This seems a bit arbitrary to me too!

@pomegranited This field is for assisting with discovery and not for the runtimes. For example, we will have a Library which lists all the bundles which are shared with the authors. One of the requirements is that authors should be able to specify the granularity of sharing, i.e. they should be able to say: "This vertical is meant to be used as a whole" or "This sequential is meant to be used as a whole". So the blocks specified in `components` are what the authors will see and be able to select for inclusion in their course and pathway.

I am going to remove the `main_component` field here. So in the data transfer command you were working on, only the root block id that was specified to the command should be listed in the `components` field.

Reply:
👍 thanks for the explanation @symbolist , I'll change the code on my data transfer PR to suit.